storage/storage.hpp

```diff
@@ -505,8 +505,14 @@ class Storage
                                                   TCountriesVec const & descendants,
                                                   MapFilesDownloader::TProgress const & downloadingMwmProgress,
                                                   TCountriesSet const & mwmsInQueue) const;
+
+  template <class ToDo>
+  void ForEachInSubtreeAndInQueue(TCountryId const & root, ToDo && toDo) const;
 };
 
+void GetQueuedCountries(Storage::TQueue const & queue, TCountriesSet & resultCountries);
+bool HasCountryId(TCountriesVec const & sorted, TCountryId const & countyId);
```

Comment:
> Why do we need these functions here, in a storage namespace? 

This method is used in two units. In storage.cpp and in storage_tests.cpp. 

> Maybe it's worth to move them to the private part of the Storage?

This function doesn't need any members of Storage. So I'd rather leave it as a free function.

Probably it's worth to adding a short comment to storage.hpp. (I'm doing that.)